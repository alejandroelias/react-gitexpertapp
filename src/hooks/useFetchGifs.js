import { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs';



export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {

        getGifs(category)
            .then(imgs => {

                setState({
                    data: imgs,
                    loading: false
                });

            })

    }, [category])

    return state;
}


//la importacion de import React from 'react', NO es necesario en este
//archivo porque esta funcion NO retorna JSX