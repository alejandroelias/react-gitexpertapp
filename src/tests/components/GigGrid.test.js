import { shallow } from 'enzyme'
import React from 'react'
import { GigGrid } from '../../components/GigGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GigGrid />', () => {
    
    const category = 'One punch man';
    // const wrapper = shallow(<GifGrid category={category}/>);

    test('should show the correct component and match with the snapshot', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GigGrid category={category}/>);
        expect(wrapper).toMatchSnapshot(); 
    });
 
    test('should change the snapshot when useFetchGifs is executed', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        },
        {
            id: '123',
            url: 'https://google.com',
            title: 'ABC Title'
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GigGrid category={category}/>);
 
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
 
    });
    
    
})
