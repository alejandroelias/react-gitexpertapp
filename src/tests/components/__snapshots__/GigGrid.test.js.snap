// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GigGrid /> should change the snapshot when useFetchGifs is executed 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch man
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
    <GifGridItem
      id="123"
      key="123"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GigGrid /> should show the correct component and match with the snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch man
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
